endpoint/endpoint.go

```diff
@@ -396,3 +397,41 @@ func RemoveDuplicates(endpoints []*Endpoint) []*Endpoint {
 
 	return result
 }
+
+// Check endpoint if is it properly formatted according to RFC standards
+func CheckEndpoint(endpoint Endpoint) bool {
+	switch recordType := endpoint.RecordType; recordType {
+	case "MX":
+		for _, target := range endpoint.Targets {
+			// MX records must have a preference value to indicate priority, e.g. "10 example.com"
+			// as per https://www.rfc-editor.org/rfc/rfc974.txt
+			targetParts := strings.Fields(strings.TrimSpace(target))
+			if len(targetParts) != 2 {
+				return false
+			}
+
+			preferenceRaw := targetParts[0]
+			_, err := strconv.ParseInt(preferenceRaw, 10, 32)
+			if err != nil {
+				return false
+			}
+		}
+	case "SRV":
+		for _, target := range endpoint.Targets {
+			// SRV records must have a priority, weight, and port value, e.g. "10 5 5060 example.com"
+			// as per https://www.rfc-editor.org/rfc/rfc2782.txt
+			targetParts := strings.Fields(strings.TrimSpace(target))
+			if len(targetParts) != 4 {
+				return false
+			}
+
+			for _, part := range targetParts[:3] {
+				_, err := strconv.ParseInt(part, 10, 32)
+				if err != nil {
+					return false
```

Comment:
shell we add debug logging here, so it's clear why record is rejected?

pseudo code
```
debug("Invalid %s record target: %s and reason %s", type, target, reason))
```